shell/plugins/steve/worker/web-worker.advanced.js

```diff
@@ -73,35 +70,61 @@ const handleConnectionError = (eventType, event, watcher) => {
 /**
  * Remove any pending messages related to this resource from the queue
  */
-const removeFromWorkerQueue = (watchKey) => {
-  state.workerQueue = state.workerQueue.filter((workerMessage) => {
+const removeFromWatcherQueue = (watchKey) => {
+  state.watcherQueue = state.watcherQueue.filter((workerMessage) => {
     const [, msg] = Object.entries(workerMessage)[0];
     const workerMessageWatchKey = watchKeyFromMessage(msg);
 
     return watchKey !== workerMessageWatchKey;
   });
 };
 
+/**
+ * Creates a resourceCache with the appropriate type
+ */
+
+const resourceCache = (type) => {
+  const CacheClass = cacheClasses[`${ type }Cache`] || cacheClasses.resourceCache;
+
+  return new CacheClass(type);
+};
+
 /**
  * These are things that we do when we get a message from the UI thread
  */
 const workerActions = {
-  // ToDo: SM we'll make a generic loader for all resource types when we need it but it'll be pretty similar to this
-  loadSchemas: (collection) => {
-    if (!caches[SCHEMA]) {
-      caches[SCHEMA] = new ResourceCache(SCHEMA);
+  createApi: () => {
+    if (!state.api) {
+      // the apiClient can only exist after schemas are loaded so we build it here.
+      state.api = new SteveClient(state.config, { loadCache: workerActions.loadCache });
+      if (!caches[SCHEMA]) {
+        state.api.request({ type: SCHEMA })
```

Comment:
I've still got to test error handling, but this is one of the place where something probably needs to happen